src/Language/PureScript/Parser/Types.hs

```diff
@@ -70,7 +70,7 @@ parseTypeAtom = indented *> P.choice
 
 parseConstrainedType :: TokenParser Type
 parseConstrainedType = do
-  constraints <- P.try (return <$> parseConstraint) <|> parens (commaSep1 parseConstraint)
+  constraints <- return <$> parseConstraint
```

Comment:
Right, and once we make it into a type operator, the thing on the left of the `=>` will also be a type, which will simplify things quite a bit.